addons/website_sale_autocomplete/static/src/interactions/address_form.js

```diff
@@ -0,0 +1,98 @@
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
+
+import { rpc } from "@web/core/network/rpc";
+import { KeepLast } from "@web/core/utils/concurrency";
+import { renderToElement } from "@web/core/utils/render";
+
+class AddressForm extends Interaction {
+    static selector = ".oe_cart .checkout_autoformat";
+    static selectorHas = "input[name='street'][data-autocomplete-enabled='1']";
+    dynamicContent = {
+        "input[name='street']": { "t-on-input": this.debounced(this.onInputStreet, 200) },
+        ".js_autocomplete_result": { "t-on-click": this.onClickAutocompleteResult },
+    };
+
+    setup() {
+        this.streetAndNumberInput = this.el.querySelector("input[name='street']");
+        this.cityInput = this.el.querySelector("input[name='city']");
+        this.zipInput = this.el.querySelector("input[name='zip']");
+        this.countrySelect = this.el.querySelector("select[name='country_id']");
+        this.stateSelect = this.el.querySelector("select[name='state_id']");
+        this.keepLast = new KeepLast();
+        this.sessionId = this.generateUUID();
+    }
+
+    generateUUID() {
+        return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
+            const r = (Math.random() * 16) | 0, v = c == "x" ? r : (r & 0x3) | 0x8;
+            return v.toString(16);
+        });
+    }
+
+    async onInputStreet(ev) {
+        const inputContainerEl = ev.currentTarget.parentNode;
+        if (ev.currentTarget.value.length >= 5) {
```

Comment:
Are you sure `currentTarget` is still available in a debouced event ? (maybe it is)